src/main/java/org/cardanofoundation/cfexploreraggregator/uniqueaccount/processor/UniqueAccountProcessor.java

```diff
@@ -66,28 +86,28 @@ public void handleCommitEvent(CommitEvent commitEvent) {
         Set<String> snapshot = new HashSet<>(uniqueAccounts);
         uniqueAccounts.clear();
         snapshot.forEach(address -> {
-            Optional<AccountInEpochEntity> byAddressAndEpoch = addressInEpochRepository.findByAddressAndEpoch(address, commitEvent.getMetadata().getEpochNumber());
-            List<AccountInEpochEntity> accountsInEpoch = new ArrayList<>();
+            Optional<UniqueAddressesInEpochEntity> byAddressAndEpoch = uniqueAddressesInEpochRepository.findByAddressAndEpoch(address, commitEvent.getMetadata().getEpochNumber());
+            List<UniqueAddressesInEpochEntity> accountsInEpoch = new ArrayList<>();
             if(byAddressAndEpoch.isEmpty()) {
-                accountsInEpoch.add(AccountInEpochEntity.builder()
+                accountsInEpoch.add(UniqueAddressesInEpochEntity.builder()
                         .address(address)
                         .epoch(commitEvent.getMetadata().getEpochNumber())
                         .slot(commitEvent.getMetadata().getSlot())
                         .build());
             }
-            addressInEpochRepository.saveAll(accountsInEpoch);
+            uniqueAddressesInEpochRepository.saveAll(accountsInEpoch);
         });
     }
 
     @EventListener
     @Transactional
     public void handleEpochChangeEvent(EpochChangeEvent epochChangeEvent) {
 
-        if(epochChangeEvent.getEpoch() == 0) {
+        if(epochChangeEvent.getPreviousEpoch() == null) {
             return;
         }
```

Comment:
wondering how rollback safe is this 😂

Reply:
Actually this is just needed for the first epoch :D